wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,194 @@
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability being provisioned.
+
+* The feature-pack contained minimum stability level is used to constrain the set of provisioned features. Any features/packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the minimum stability level of each feature-pack is used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different minimum stability level. 
+The minimum stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a stability-level, by mean of the `stability-level` Galleon option. 
+This stability level is used to constrain the stability level of all provisioned features/packages. If the specified level is lower than a feature-pack minimum stability, 
+the feature-pack minimum stability is used.
+
+For example, a provisioning configuration containing 3 feature-packs:
+
+* A, minimum stability default
+* B, minimum stability community
+* C, minimum stability experimental
+
+If no stability level is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+If the stability level `default` is specified:
+
+* `default` features/packages of A are provisioned
+* `default` features/packages of B are provisioned
+* `default` features/packages of C are provisioned
+
+If the stability level `experimental` is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+==== Lowest stability level needed by WildFly Galleon Plugins
+
+At provisioning time, when WildFly Galleon plugin generates the server configurations, 
+a WildFly embedded server is started. This server is started with a stability level computed in the following way:
+
+* If a stability level has been specified by the user, this level is used.
+* If no stability level has been specified, the lowest minimum stability level of all feature-packs present in the provisioning is used.
+
+=== Provisioning tooling requirement
+
+* All Galleon based tooling should let users specify the `stability-level` option.
+* In addition, for WildFly Glow, information about the features/packages that would be not provisioned for a given stability level 
+is displayed to the user.
+
+=== Nice-to-Have Requirements
+
+* NONE
+
+=== Non-Requirements
+
+* Recording and utilizing stability information in Galleon configuration elements 
+at a higher granularity level than a feature-spec or package (i.e. configs, layers, feature-groups).
```

Comment:
@jfdenise You somewhat cover this with your use of 'utilizing', but, given your comment about 2) at https://github.com/wildfly/wildfly-proposals/pull/555/files#r1502690469 above, I suggest adding:

* If a provisioning plan (i.e. a provisioning.xml or other provisioning tooling configuration that instructs the tooling about what to provision) involves configs, layers of feature-groups that reference features or packages that are not available at the stability level in effect, the result is not specified. That is:
** Always rejecting provisioning such a server is not required.
** Always provisioning a working server is not required.
** Best practice for feature pack authors is to not include features or packages at a lower stability level in configs, layers or feature-groups that are expected to be used in servers provisioned at a higher stability level. However it is not the responsibility of the Galleon tooling to enforce this best practice or to guarantee a particular result if best practice is not followed.

We could say more about how the features/packages that can be provisioned at the given stability level will be provisioned, and then the overall provisioning and the resulting installation may or may not work. But since the end result is it may or may not work, I don't think we should discuss how 'features/packages that can be provisioned at the given stability level will be'. At least not in this iteration.

Reply:
Great phrasing. Included!